**Changes Requested**

The implementation approach is sound -- threading `edge_clearance` through `auto_pour_if_missing` to `ZoneGenerator` and using Shapely's `buffer(-clearance)` for polygon inset is the right design. The code in `generator.py` correctly handles the fallback when shapely is unavailable (catches `ImportError`, warns, falls back to exact outline), and explicit boundaries are properly left untouched. The `_inset_polygon` static method handles edge cases well (collapsed polygons, MultiPolygon results from thin regions).

However, there is one issue that needs to be fixed:

**`tests/test_auto_pour.py` -- missing shapely guard on new tests**

The two new tests added to `test_auto_pour.py` (`test_edge_clearance_insets_zone_boundary` at line 210 and `test_no_edge_clearance_uses_exact_outline` at line 256) do not guard against missing shapely. Shapely is an *optional* dependency under `[project.optional-dependencies] geometry`, not part of the core or `dev` extras. The CI workflow installs with `uv sync --extra dev`, which does not include shapely.

- `test_edge_clearance_insets_zone_boundary` will **fail** when shapely is not installed because the `ImportError` fallback in `board_outline` means zones use the exact board outline, causing the coordinate assertions to fail (observed locally: `AssertionError: X coord 0.0 too close to left edge`).
- `test_no_edge_clearance_uses_exact_outline` happens to pass without shapely (since no inset is applied regardless), but it would be better to be consistent.

Compare with `tests/test_zone_generator_edge_clearance.py` which correctly uses `shapely = pytest.importorskip("shapely", reason="shapely required for edge clearance tests")` at the module level.

**Fix**: Add `pytest.importorskip("shapely", reason="shapely required for edge clearance tests")` to `test_edge_clearance_insets_zone_boundary`. The second test (`test_no_edge_clearance_uses_exact_outline`) does not strictly require it but could include the guard for consistency.

Everything else looks good:
- The four call sites in `route_cmd.py` are updated consistently
- The `auto_pour_if_missing` and `auto_create_zones_for_pour_nets` parameter threading is clean
- `ZoneGenerator.__init__` and `from_pcb` properly accept and forward `edge_clearance`
- The `_inset_polygon` helper is well-implemented with proper MultiPolygon and collapsed-polygon handling
- `test_zone_generator_edge_clearance.py` has thorough coverage with 7 tests
- PR description properly includes `Closes #2411`

--- a/tests/test_auto_pour.py
+++ b/tests/test_auto_pour.py
@@ -54,9 +54,7 @@ def _make_pcb(
         parts.append(f'  (net {nid} "{name}")\n')
 
     # Single dummy footprint with pads
-    parts.append(
-        '  (footprint "TestLib:TestPkg" (layer "F.Cu") (at 10 10)\n'
-    )
+    parts.append('  (footprint "TestLib:TestPkg" (layer "F.Cu") (at 10 10)\n')
     for idx, (nid, name) in enumerate(pad_nets):
         x_off = idx * 2.0
         parts.append(
@@ -209,6 +207,9 @@ class TestAutoPourIfMissing:
 
     def test_edge_clearance_insets_zone_boundary(self, tmp_path: Path):
         """Zone boundary is inset from board edge when edge_clearance is set."""
+        pytest.importorskip(
+            "shapely", reason="shapely required for edge clearance tests"
+        )
         import re
 
         from kicad_tools.router.auto_pour import auto_pour_if_missing
@@ -220,9 +221,7 @@ class TestAutoPourIfMissing:
         pcb_path = tmp_path / "test.kicad_pcb"
         pcb_path.write_text(pcb)
 
-        count, names = auto_pour_if_missing(
-            pcb_path, edge_clearance=0.3
-        )
+        count, names = auto_pour_if_missing(pcb_path, edge_clearance=0.3)
 
         assert count == 1
         assert names == ["GND"]
@@ -230,9 +229,7 @@ class TestAutoPourIfMissing:
         # Parse zone polygon coordinates from the written file
         text = pcb_path.read_text()
         # Extract xy coordinates from the zone polygon
-        xy_matches = re.findall(
-            r"\(xy\s+([\d.e+-]+)\s+([\d.e+-]+)\)", text
-        )
+        xy_matches = re.findall(r"\(xy\s+([\d.e+-]+)\s+([\d.e+-]+)\)", text)
         assert len(xy_matches) > 0, "No zone polygon coordinates found"
 
         # Board edge is 0..50 in both X and Y (from _PCB_FOOTER).
@@ -240,21 +237,24 @@ class TestAutoPourIfMissing:
         # at least 0.3mm inward from the edges.
         for x_str, y_str in xy_matches:
             x, y = float(x_str), float(y_str)
-            assert x >= 0.3 - 0.01, (
-                f"X coord {x} too close to left edge (expected >= 0.3)"
-            )
-            assert x <= 49.7 + 0.01, (
-                f"X coord {x} too close to right edge (expected <= 49.7)"
-            )
-            assert y >= 0.3 - 0.01, (
-                f"Y coord {y} too close to top edge (expected >= 0.3)"
-            )
-            assert y <= 49.7 + 0.01, (
-                f"Y coord {y} too close to bottom edge (expected <= 49.7)"
-            )
+            assert (
+                x >= 0.3 - 0.01
+            ), f"X coord {x} too close to left edge (expected >= 0.3)"
+            assert (
+                x <= 49.7 + 0.01
+            ), f"X coord {x} too close to right edge (expected <= 49.7)"
+            assert (
+                y >= 0.3 - 0.01
+            ), f"Y coord {y} too close to top edge (expected >= 0.3)"
+            assert (
+                y <= 49.7 + 0.01
+            ), f"Y coord {y} too close to bottom edge (expected <= 49.7)"
 
     def test_no_edge_clearance_uses_exact_outline(self, tmp_path: Path):
         """Without edge_clearance, zone boundary matches board edge exactly."""
+        pytest.importorskip(
+            "shapely", reason="shapely required for edge clearance tests"
+        )
         import re
 
         from kicad_tools.router.auto_pour import auto_pour_if_missing
@@ -272,9 +272,7 @@ class TestAutoPourIfMissing:
         text = pcb_path.read_text()
         # Zone polygon should include coordinates at or very near the
         # board edge (0 and 50).
-        xy_matches = re.findall(
-            r"\(xy\s+([\d.e+-]+)\s+([\d.e+-]+)\)", text
-        )
+        xy_matches = re.findall(r"\(xy\s+([\d.e+-]+)\s+([\d.e+-]+)\)", text)
         xs = [float(x) for x, _ in xy_matches]
         ys = [float(y) for _, y in xy_matches]
 

